examples/src/test/scala/tools/mongoimport.scala

```diff
@@ -0,0 +1,221 @@
+                |  --file arg                    file to import from; if not specified stdin is used
+                |  --drop                        drop collection first
+              """.stripMargin
+
+  /**
+   * The main export program
+   * @param args the commandline arguments
+   */
+  def main(args: Array[String]) {
+
+    // The time when the execution of this program started, in milliseconds since 1 January 1970 UTC.
+    val executionStart: Long = currentTime
+
+
+    if (args.length == 0 | args.contains("--help")) {
+      Console.err.println(usage)
+      sys.exit(1)
+    }
+
+    // Set the debug log level
+    java.util.logging.Logger.getLogger("").getHandlers.foreach(h => h.setLevel(Level.WARNING))
+
+    val optionMap = parseArgs(Map(), args.toList)
+    val options = getOptions(optionMap)
+
+    if (options.uri.isEmpty) {
+      Console.err.println(s"Missing URI")
+      Console.err.println(usage)
+      sys.exit(1)
+    }
+
+    // Get source
+    val importSource: BufferedSource = options.file match {
+      case None => Source.stdin
+      case Some(fileName) => Source.fromFile(fileName)
+    }
+
+    // Get URI
+    val mongoClientURI = new ConnectionString(options.uri.get)
+    if (Option(mongoClientURI.getCollection).isEmpty) {
+      Console.err.println(s"Missing collection name in the URI eg:  mongodb://<hostInformation>/<database>.<collection>[?options]")
+      Console.err.println(s"Current URI: $mongoClientURI")
+      sys.exit(1)
+    }
+
+    // Get the collection
+    val mongoClient = MongoClient(mongoClientURI.getURI)
+    val collection = mongoClient.getDatabase(mongoClientURI.getDatabase).getCollection(mongoClientURI.getCollection)
+
+    if (options.drop) {
+      if (!options.quiet) Console.err.println(s"Dropping: ${mongoClientURI.getCollection}")
+      Await.result(collection.drop().head(), Duration(10, TimeUnit.SECONDS))
+    }
+
+    if (!options.quiet) Console.err.print("Importing...")
+    val importer = importJson(collection, importSource.getLines(), Promise[Completed]()).future
+    showPinWheel(importer)
+    importSource.close()
+    val total = currentTime - executionStart
+    if (!options.quiet) Console.err.println(s"Finished import: $total ms")
+  }
+
+  /**
+   * Imports JSON into the collection
+   *
+   * @param collection the mongodb collection to insert into
+   * @param lines the iterator from the importSource
+   * @param promise the promise that is fufilled on completion or on error
+   * @return the promise
+   */
+  private def importJson(collection: MongoCollection[Document], lines: Iterator[String], promise: Promise[Completed]): Promise[Completed] = {
+    lines.hasNext match {
+      case true => collection.insertMany(lines.take(1000).map(json => Document(json)).toSeq).subscribe(
+        (completed: Completed) => lines.hasNext match {
+            case true => importJson(collection, lines, promise)
```

Comment:
The docs for Iterator.take say:

     Reuse: After calling this method, one should discard the iterator it was called on, and use only the iterator that was returned. Using the old iterator is undefined, subject to change, and may result in changes to the new iterator as well

Not sure if that's an issue here or not.